### PR TITLE
Add adaptive soft link for /dev/sgx

### PR DESCRIPTION
**Pull request by Uxito-Ada**
Enable this can contribute to avoid SGX open failure during k8s start, which may be caused by a wrong mount.

### CODE REVIEW

**Comment by yang8621** [comment]
Could merge the three into one commit?

File: docker/core/start_ehsm-kms.sh [hunk 1]
```diff
@@ -4,4 +4,26 @@ echo 'PCCS_URL='${PCCS_URL}'/sgx/certification/v3/' >> /etc/sgx_default_qcnl.con
...
+fi
+
+if [ -c "/dev/sgx/provision" ]; then
+    echo "/dev/sgx/provision is ready"
+elif [ -c "/dev/sgx_provision" ]; then
+    echo "/dev/sgx/provision not ready, try to link to /dev/sgx_provision"
+    mkdir -p /dev/sgx
+    ln -s /dev/sgx_provision /dev/sgx/provision
+else
+    echo "both /dev/sgx/provision /dev/sgx_provision are not ready, please check the kernel and driver"
```
Comment (yang8621) [inline docker/core/start_ehsm-kms.sh:24]: Neither .. nor ..

Comment (yang8621) [reply docker/core/start_ehsm-kms.sh:24]: exit 1

**Review by syan10 — COMMENTED** [review]
please combine the 3 commits into one.

**Comment by Uxito-Ada** [comment]
Commits are squashed into one.

File: docker/core/start_ehsm-kms.sh (hunk 1 above)
Comment (Uxito-Ada) [reply docker/core/start_ehsm-kms.sh:25]: Add exit 1 for both checks fail

Comment (Uxito-Ada) [reply docker/core/start_ehsm-kms.sh:25]: Add exit 1 for both checks fail

**Review by yang8621 — APPROVED** [review]

**Comment by syan10** [comment]
you can fix it by modify the yaml file, so close it.

**Pull request closed without merging**